cloud/pkg/common/informers/informers.go

```diff
@@ -0,0 +1,105 @@
+package informers
+
+import (
+	"sync"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	k8sinformer "k8s.io/client-go/informers"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+
+	crdinformers "github.com/kubeedge/kubeedge/cloud/pkg/client/informers/externalversions"
+	"github.com/kubeedge/kubeedge/cloud/pkg/common/client"
+)
+
+type newInformer func() cache.SharedIndexInformer
+
+type KubeEdgeCustomeInformer interface {
+	EdgeNode() cache.SharedIndexInformer
+}
+
+type Manager interface {
+	GetK8sInformerFactory() k8sinformer.SharedInformerFactory
+	GetCRDInformerFactory() crdinformers.SharedInformerFactory
+	KubeEdgeCustomeInformer
+	Start(stopCh <-chan struct{})
+}
+
+var globalInformers Manager
+var once sync.Once
+
+func GetInformersManager() Manager {
+	once.Do(func() {
+		globalInformers = &informers{
+			defaultResync:            0,
+			keClient:                 client.GetKubeEdgeClient(),
+			informers:                make(map[string]cache.SharedIndexInformer),
+			startedInformers:         make(map[string]bool),
+			crdSharedInformerFactory: crdinformers.NewSharedInformerFactory(client.GetKubeEdgeClient(), 0),
+			k8sSharedInformerFactory: k8sinformer.NewSharedInformerFactory(client.GetKubeEdgeClient(), 0),
+		}
+	})
+	return globalInformers
+}
+
+type informers struct {
+	defaultResync            time.Duration
+	keClient                 client.KubeEdgeClient
+	lock                     sync.Mutex
+	informers                map[string]cache.SharedIndexInformer
+	startedInformers         map[string]bool
```

Comment:
We already have the informerfactory, so do we still need these maps?

Reply:
The custome Informers stored in the informers field also need to run only once. There will have some problem if we run those informers more than one times, so i think we need it.

Reply:
For the pod, [`Informer`](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/informers/core/v1/pod.go#L84) here will use the [`InformerFor`](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/informers/factory.go#L169) to create the sharedinformer, and if the informer existed, it will return directly, so will it run many times?

Reply:
done!